mantidproject/mantid
# Engineering Diffraction GUI save sample logs to HDF5 after focusing

A new algorithm `ExportSampleLogsToHDF5` was implemented to do what it says on the tin. This algorithm was integrated into the **Focus** tab of the ED GUI - the GUI should now save all the sample logs to a HDF5 file after focusing a run.

**To test:**

There is some sample data [here](https://github.com/mantidproject/mantid/files/2057704/PR22357.zip).

If you haven't used the GUI before, you'll be prompted to enter an RB number - just put anything in the box. Before you do a focus, you'll have to run a calibration. Go to the Calibration tab, select the 280329 file for Vanadium and 280336 for sample, and click Calibrate. If you're in debug mode go and make a cup of coffee.

Now go to the Focus tab and select the the 281188 file for Run, then click Focus. Once this has completed, go to your output directory (on Windows this will be `C:\EnginX_Mantid\User\<rb_number>\Runs`, on Linux `~/EnginX_mantid/User/<rb_number>/Runs`). `281188.hdf5` should be there. Have a look at it with your favourite HDF5 viewer, and make sure that the sample logs for the run are present. Time-series logs should have been averaged and properties with empty values should have been skipped

Using some data of your own, have a play with `ExportSampleLogsToHDF5` and make sure everything looks sensible

Fixes #21446 and #17168 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/pull/22357/commits/7876182fbecb6843f7d6fddfdbb6a8556435923d)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
